File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // if(nums.length == 2){
        //     return new int[]{0, 1};
        // }

        // for(int i =0; i<nums.length; i++){
        //     for(int j=i+1; j<nums.length;j++){
        //         if(nums[i]+nums[j]==target && i!=j){
        //             return new int[]{i, j};
        //         }
        //     }
        // }
        // return new int[]{0};
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i =0; i<nums.length;i++){
            int count = target-nums[i];
            if(map.containsKey(count)){
                return new int[]{map.get(count),i};
            }else{
                map.put(nums[i],i);
            }
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums)-1
        
        while(l<r):
            mid = int((l+r)/2)
            if(nums[mid] < nums[r]):
                r = mid
            else:
                l = mid + 1
        pivot = l
        if(nums[l] <= target and target <= nums[-1]):
            return self.binarySearch(nums, pivot,len(nums)-1,target)
        else:
            return self.binarySearch(nums, 0, pivot-1,target)
    def binarySearch(self, nums, l, r, target):
        while(l<=r):
            mid = int((l+r)/2)
            if(nums[mid] == target):
                return mid
            elif(nums[mid]<target):
                l = mid + 1
            else:
                r = mid-1
        return -1